base_layer/wallet/src/output_manager_service/service.rs

```diff
@@ -1888,97 +1891,159 @@ where
-                        &blinding_factor,
-                        committed_value.into(),
-                    )? {
-                        let rewound_output = UnblindedOutput::new(
-                            output.version,
-                            committed_value,
-                            blinding_factor.clone(),
-                            output.features,
-                            known_one_sided_payment_scripts[i].script.clone(),
-                            known_one_sided_payment_scripts[i].input.clone(),
-                            known_one_sided_payment_scripts[i].private_key.clone(),
-                            output.sender_offset_public_key,
-                            output.metadata_signature,
-                            known_one_sided_payment_scripts[i].script_lock_height,
-                            output.covenant,
-                            output.encrypted_value,
-                        );
-
-                        let db_output = DbUnblindedOutput::rewindable_from_unblinded_output(
-                            rewound_output.clone(),
-                            &self.resources.factories,
-                            &RewindData {
-                                rewind_blinding_key,
-                                encryption_key,
-                            },
-                            None,
-                            Some(&output.proof),
-                        )?;
-
-                        let output_hex = output.commitment.to_hex();
-                        let tx_id = TxId::new_random();
-
-                        match self.resources.db.add_unspent_output_with_tx_id(tx_id, db_output) {
-                            Ok(_) => {
-                                rewound_outputs.push(RecoveredOutput {
-                                    output: rewound_output,
-                                    tx_id,
-                                });
-                            },
-                            Err(OutputManagerStorageError::DuplicateOutput) => {
-                                warn!(
-                                    target: LOG_TARGET,
-                                    "Attempt to add scanned output {} that already exists. Ignoring the output.",
-                                    output_hex
-                                );
-                            },
-                            Err(err) => {
-                                return Err(err.into());
+        let scanned_outputs = outputs
+            .into_iter()
+            .filter_map(|output| {
+                // TODO: use MultiKey
+                match output.script.as_slice() {
+                    // ----------------------------------------------------------------------------
+                    // simple one-sided address
+                    [Opcode::PushPubKey(spending_key_public)] => {
+                        let keys = match self.resources.db.get_all_known_one_sided_payment_scripts() {
```

Comment:
Calling this db method for every output is not necessary, rather move it outside the loop